ui/src/services/liquidity.service.js

```diff
@@ -7,77 +7,24 @@ import {
   calcValueToRemove,
 } from '@agoric/zoe/src/contractSupport';
 
-import { dappConfig } from '../utils/config.js';
-
-const createNewPurse = async (
-  liquidityBrand,
+export const createNewPurse = async (
   walletP,
-  instanceID,
-  contractName,
-) => {
-  console.log(instanceID, contractName);
-  const board = await E(walletP).getBoard();
-  const zoe = await E(walletP).getZoe();
-  const instance = await E(board).getValue(instanceID);
-  const issuers = await E(zoe).getIssuers(instance);
-  const liquidityBrandName = await E(liquidityBrand).getAllegedName();
-
-  const liquidityIssuer = issuers[liquidityBrandName];
+  secondaryBrandName,
+  liquidityIssuerId,
+) =>
+  E(walletP).suggestIssuer(`${secondaryBrandName}Liquidity`, liquidityIssuerId);
 
-  if (!liquidityIssuer) {
-    throw Error('Liquidity issuer not found in AMM');
-  }
-  const liquidityId = await E(board).getId(liquidityIssuer);
-  await E(walletP).suggestIssuer(liquidityBrandName, liquidityId);
-
-  // purseName of newly created purse will come out as array
-  const newName = [];
-  newName.push(contractName);
-  newName.push(liquidityBrandName);
-
-  return newName;
-};
-
-export const addLiquidityService = async (
+export const addLiquidityService = (
   centralAmount,
   centralPurse,
   secondaryAmount,
   secondaryValuePurse,
-  ammAPI,
+  liquidityPurse,
   walletP,
-  purses,
   pool,
 ) => {
-  const { AMM_INSTALLATION_BOARD_ID, AMM_INSTANCE_BOARD_ID, CONTRACT_NAME } =
-    dappConfig;
-  const liquidityBrand = pool.liquidityTokens.brand;
-  let liquidityPurse = purses.find(purse => purse.brand === liquidityBrand);
-
-  if (liquidityPurse) {
-    liquidityPurse = liquidityPurse.pursePetname;
-  } else {
-    liquidityPurse = await createNewPurse(
-      liquidityBrand,
-      walletP,
-      AMM_INSTANCE_BOARD_ID,
-      CONTRACT_NAME,
-    );
-  }
-
-  const id = `${Date.now()}`;
-  let invitation;
-  try {
-    invitation = await E(ammAPI).makeAddLiquidityInvitation();
-  } catch (error) {
-    return {
-      status: 500,
-      message:
-        error ||
-        error.message ||
-        error.error ||
-        'Something went wrong while creating invitation for add liquidity',
-    };
-  }
+  // TODO: Get this a better way?
```

Comment:
more of a FIXME imo, and this should reference a ticket to solve it

Reply:
#5959

Reply:
Done